### PR TITLE
feat(GraphQL): Reduce query complexity

### LINKED ISSUE #70: Use GraphQL CodeGen to interrogate Contentful and automatically generate TS types
See comment here: https://github.com/Royal-Navy/docs.royalnavy.io/pull/67/files#r593061461

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related issue

Closes #70, #73

## Overview

Reduce query complexity; use `graphql-codegen` and `fetch` for client.

## Reason

>We need introspection to generate TS types from the GraphQL schema and we need to make multiple queries to sit within the Contentful query complexity limit.

## Work carried out

- [x] Simplify content model
- [x] Move adapters into their own directory
- [x] Use graphql-codegen for type generation and fetch for GraphQL client
- [x] Use multiple queries to stay within Contenful complexity limit

## Developer notes

We can't use Apollo because it relies on using hooks throughout the component tree.

NextJS requires all data for SSG to be prefetched at the top of the component tree by `getStaticProps`.

https://github.com/vercel/next.js/discussions/11957

### CODE REVIEW

**Comment by m7kvqbe1** [comment]
Some of the typing still needs tightening up but it's a bit of a pain to be honest.

Mainly because we're wrangling the data a bit in `getStaticProps` with the async stuff.

It's really quite frustrating hitting the query complexity limit and having to fire off multiple queries.

File: pages/index.tsx [hunk 1]
```diff
@@ -23,71 +26,37 @@ interface HomeProps {
...
-                  }
-                }
-              }
-            }
-          }
-        }
-      }
+export const getStaticProps: GetStaticProps = async () => {
+  const response = await fetchContent(PAGE_STRUCTURE_QUERY, {
+    id: '7ltwNe3eIZtPNZ2xrHMAuw',
```
Comment (m7kvqbe1) [inline pages/index.tsx:31]: Not happy about this hard-coded ID. Maybe we can infer this from a route slug.

File: babel.config.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ module.exports = {
     '@babel/plugin-proposal-object-rest-spread',
     '@babel/plugin-proposal-nullish-coalescing-operator',
     '@babel/plugin-proposal-optional-chaining',
+    'import-graphql',
```
Comment (m7kvqbe1) [inline babel.config.js:8]: This automatically parses a .graphql import into an AST for use. 

It does the same thing as this:

https://github.com/apollographql/graphql-tag

**Review by thyhjwb6 — APPROVED** [review]
👍

**Pull request merged**